src/domain/service_domain_db.erl

```diff
@@ -0,0 +1,141 @@
+              }}.
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+enabled() ->
+    is_pid(whereis(?MODULE)).
+
+force_check_for_updates() ->
+    %% Send a broadcast message.
+    case pg2:get_members(?GROUP) of
+        Pids when is_list(Pids) ->
+            [Pid ! check_for_updates || Pid <- Pids],
+            ok;
+        {error, _Reason} -> ok
+    end.
+
+%% Does nothing but blocks until every member processes its queue.
+sync() ->
+    case pg2:get_members(?GROUP) of
+        Pids when is_list(Pids) ->
+            [gen_server:call(Pid, ping) || Pid <- Pids],
+            ok;
+        {error, _Reason} -> ok
+    end.
+
+%% ---------------------------------------------------------------------------
+%% Server callbacks
+
+init([]) ->
+    pg2:create(?GROUP),
+    pg2:join(?GROUP, self()),
+    LastEventId = initial_load(mongoose_domain_core:get_last_event_id()),
+    ?LOG_INFO(#{what => domains_loaded, last_event_id => LastEventId}),
+    State = #{last_event_id => LastEventId,
+              check_for_updates_interval => 30000},
+    {ok, handle_check_for_updates(State)}.
+
+handle_call(ping, _From, State) ->
+    {reply, pong, State};
+handle_call(Request, From, State) ->
+    ?UNEXPECTED_CALL(Request, From),
+    {reply, ok, State}.
+
+handle_cast(Msg, State) ->
+    ?UNEXPECTED_CAST(Msg),
+    {noreply, State}.
+
+handle_info(check_for_updates, State) ->
+    {noreply, handle_check_for_updates(State)};
+handle_info(Info, State) ->
+    ?UNEXPECTED_INFO(Info),
+    {noreply, State}.
+
+terminate(_Reason, _State) ->
+    ok.
+
+code_change(_OldVsn, State, _Extra) ->
+    {ok, State}.
+
+%% ---------------------------------------------------------------------------
+%% Server helpers
+
+initial_load(undefined) ->
+    link(whereis(mongoose_domain_core)),
+    LastEventId = mongoose_domain_sql:get_max_event_id(),
+    PageSize = 10000,
+    mongoose_domain_loader:load_data_from_base(0, PageSize),
+    mongoose_domain_core:set_last_event_id(LastEventId),
+    unlink(whereis(mongoose_domain_core)),
+    LastEventId;
+initial_load(LastEventId) when is_integer(LastEventId) ->
+    LastEventId. %% Skip initial init
+
+handle_check_for_updates(State = #{last_event_id := LastEventId,
+                                   check_for_updates_interval := Interval}) ->
+    maybe_cancel_timer(State),
+    receive_all_check_for_updates(),
+    PageSize = 1000,
+    LastEventId2 = mongoose_domain_loader:check_for_updates(LastEventId, PageSize),
+    maybe_set_last_event_id(LastEventId, LastEventId2),
+    TRef = erlang:send_after(Interval, self(), check_for_updates),
+    State#{last_event_id => LastEventId2, check_for_updates => TRef}.
+
+maybe_cancel_timer(#{check_for_updates_tref := TRef}) ->
+    erlang:cancel_timer(TRef);
+maybe_cancel_timer(_) ->
+    ok.
+
+receive_all_check_for_updates() ->
+    receive check_for_updates -> ok after 0 -> ok end.
```

Comment:
Wasn't this supposed to receive all `check_for_updates` messages? It receives just one.